cogito/putter.go

```diff
@@ -83,14 +98,37 @@ func (putter *ProdPutter) ProcessInputDir() error {
 
 	params := putter.Request.Params
 	source := putter.Request.Source
+
+	sinks := sets.From(source.Sinks...)
+	if len(params.Sinks) > 0 {
+		sinks = sets.From(params.Sinks...)
+	}
+
 	var msgDir string
 
+	if len(putter.InputDir) == 0 && (sinks.Size() > 0 && !sinks.Contains("github")) {
+		// Nothing to validate, InputDir is not mandatory if only chat feature is requested.
+		// Later, will repeat the check for collected directories.
+		return nil
+	}
 	collected, err := collectInputDirs(putter.InputDir)
 	if err != nil {
 		return err
 	}
 
 	inputDirs := sets.From(collected...)
+	// If putter.InputDir is not set, it is likely only Cogito chat feature is requested.
+	// Sinks can be set as Source, let check if they are configured and 'github' keyword
+	// is present. Also we return an error if 'sinks' is not configured in this case.
+	if inputDirs.Size() == 0 && (sinks.Size() == 0 || sinks.Contains("github")) {
+		return fmt.Errorf(
+			"put:inputs: missing directory for GitHub repo: have: %v, GitHub: %s/%s",
+			inputDirs, source.Owner, source.Repo)
+	}
+	// No errors, inputDirs is not mandatory at this point.
+	if inputDirs.Size() == 0 && (sinks.Size() > 0 && !sinks.Contains("github")) {
```

Comment:
why do you need this check twice?

Reply:
I will simplify grouping the common part that is `inputDirs.Size() == 0`.
 The first check, analyzes the `put` `inputs` and if there are no inputs you are in 2 possible scenarios:
1. if sinks are not specified or they are specified and `github` id found, this is an error, inputs are missing
2. if sinks are specified and `github` is not found, then it's ok we can return.